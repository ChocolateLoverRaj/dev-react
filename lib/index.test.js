/* global test, expect */

import DevReact from './index.js'

test('constructor', () => {
  const devReact = new DevReact('dir')
  expect(devReact).toBeInstanceOf(DevReact)
})

describe('errors', () => {
  test('no new', () => {
    expect(() => {
      DevReact()
    }).toThrowErrorMatchingSnapshot()
  })

  test('bad inputDir', () => {
    expect(() => {
      new DevReact()
    }).toThrowErrorMatchingSnapshot()
    expect(() => {
      new DevReact(69)
    }).toThrowErrorMatchingSnapshot()
    expect(() => {
      new DevReact({})
    }).toThrowErrorMatchingSnapshot()
  })
})

