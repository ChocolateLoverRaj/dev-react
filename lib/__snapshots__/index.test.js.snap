// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errors bad inputDir 1`] = `"Error normalizing inputDir: TypeError: Cannot read property 'replace' of undefined"`;

exports[`errors bad inputDir 2`] = `"Error normalizing inputDir: TypeError: p.replace is not a function"`;

exports[`errors bad inputDir 3`] = `"Error normalizing inputDir: TypeError: p.replace is not a function"`;

exports[`errors no new 1`] = `"Class constructor DevReact cannot be invoked without 'new'"`;
